### PR TITLE
NXP-29107: make explorer ftest using new window more robust

### PR DESCRIPTION
**Pull request by atchertchian**
Try to avoid random failure for new tests.

T&P at https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master/217/ passed but did not create a PR because of merging conflict (linked to other changes merged in the meantime).

Launched again (without unit tests) at https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master/218/ just in case it still catches a random failure.

### CODE REVIEW

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java [hunk 1]
```diff
@@ -196,9 +202,27 @@ protected void storeWindowHandle() {
     }
 
     protected void switchToNewWindow() {
-        for (String winHandle : driver.getWindowHandles()) {
-            driver.switchTo().window(winHandle);
+        Wait<WebDriver> wait = Locator.getFluentWait();
+        wait.until((new Function<WebDriver, Boolean>() {
```
Comment (efge) [inline ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:206]: Can't a lambda be used here?

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java [hunk 2]
```diff
@@ -196,9 +202,27 @@ protected void storeWindowHandle() {
...
+                }
+            }
+        }));
+        // select last handle
+        String handle = null;
+        Iterator<String> handleIt = driver.getWindowHandles().iterator();
+        while (handleIt.hasNext()) {
+            handle = handleIt.next();
         }
+        System.err.println(driver.getWindowHandles());
```
Comment (efge) [inline ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:223]: Remove debug println

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java [hunk 3]
```diff
@@ -196,9 +202,27 @@ protected void storeWindowHandle() {
...
+                } catch (StaleElementReferenceException e) {
+                    return null;
+                }
+            }
+        }));
+        // select last handle
+        String handle = null;
+        Iterator<String> handleIt = driver.getWindowHandles().iterator();
+        while (handleIt.hasNext()) {
+            handle = handleIt.next();
```
Comment (efge) [inline ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:221]: You can use

    for (String h : driver.getWindowHandles()) {
        handle = h;
    }

Or if you feel fancy

    String handle = driver.getWindowHandles().stream().reduce((prev, next) -> next).orElse(null);

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java (hunk 2 above)
Comment (atchertchian) [reply ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:223]: oops, thanks

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java (hunk 3 above)
Comment (atchertchian) [reply ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java]: :+1: i won't be fancy for clarity :)

**Comment by atchertchian** [comment]
Relaunched again https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master/219

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java (hunk 1 above)
Comment (atchertchian) [reply ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:206]: Maybe, but we need to cleanup these logics in Locator then (where we use this everywhere so i took example on it).
I can clean this up later if needed.

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java [hunk 4]
```diff
@@ -24,6 +24,7 @@
 import java.io.IOException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
```
Comment (efge) [inline ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:27]: Import not needed anymore

File: ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java (hunk 1 above)
Comment (efge) [reply ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java:206]: Actually we can't use a lambda as `com.google.common.base.Function` is not a SAM-type (single abstract method) required for lambda use despite being annotated `@FunctionalInterface`

**Comment by atchertchian** [comment]
new T&P after Jenkins crash: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master/220/

**Pull request merged**